pkg/exec/exec.go

```diff
@@ -27,36 +28,59 @@ type OSExecutor struct {
 	DataPath string
 }
 
-const loggerKey = 1
-
 func GetLogger(ctx context.Context) log.Logger {
-	if logger, ok := ctx.Value(loggerKey).(*log.Logger); ok {
+	if logger, ok := ctx.Value(middleware.LoggerKey).(*log.Logger); ok {
 		return *logger
 	}
 
+	// Only a programming bug should cause this scenario, but the addition of
+	// this error message should at least allow us to spot a regression.
+	log.Error("Unable to retrieve logger from context, log lines will be missing")
```

Comment:
If this is a programming error, can we panic? Would that make it far more obvious, without being something that could cause us problems in production?

Reply:
Fair point - I originally thought that killing the whole server was a bit overkill and wouldn't be a nice experience for the client, but given that this should never happen (again) I think it's fair and definitely gives us a better chance of spotting a problem.

Have changed this to use `log.Fatal`